cmd/axtSam/axtSam.go

```diff
@@ -52,15 +52,21 @@ func main() {
 }
 
 func axtToSam(axtfile string, header *sam.SamHeader, output string) {
-	writer := fileio.EasyCreate(output)
+	reader, writer := fileio.EasyOpen(axtfile), fileio.EasyCreate(output)
 
 	defer writer.Close()
 
 	//setting up channels and wait groups
 	data, results := make(chan axt.Axt, 824), make(chan *sam.SamAln, 824)
 	var working, writingJob sync.WaitGroup
 
-	go axt.ReadToChan(axtfile, data)
+	var wg sync.WaitGroup
+	go axt.ReadToChan(reader, data, &wg)
+
+	go func() {
+		wg.Wait()
```

Comment:
This may actually be an example of the added complexity being difficult where we have now removed the file close from ReadToChan, but it is also not included in the waiting go routine.